added list mode for display

theonesean opened the pull request:
* added `displayMode` preference
* `displayMode=list` will display tasks in a single column list
* default is `displayMode=table`
* closes #12, with the exception of `listModePreserveWidth`, which is not implemented yet.
* all documentation updated
* need to see if `displayMode` can be overwritten selector-by-selector, or if that functionality needs implemented
* **also:** updated makefile to better support mac installation

fennifith commented on source/print.d:
```diff
@@ -57,14 +61,27 @@ class Printer {
 
     string[] printList(List list) {
+
+      if (0 == cmp(this.displayMode, "list")) {
```
Could this just be `this.displayMode == "list"`? D is not C, luckily - I think this works here.

[`string.cmp`](https://dlang.org/library/std/string.html) internally seems to reference [`std.algorithm.comparison.cmp`](https://dlang.org/library/std/algorithm/comparison/cmp.html) which just seems strange to use as an equality check. Unless you're using C, in which case it makes perfect sense, because it's C.

fennifith commented on source/print.d:
```diff
@@ -84,21 +101,52 @@ class Printer {
+
+    	foreach (x, list; lists) {
+        string[] rows;
+        if (isList) {
```
This could be simplified to a ternary operator (again, very glad that this is not C):
`string[] rows = isList ? printList(list) : printList(list, size)`

theonesean replied:
i get to use a TERNARY OPERATOR! my favorite!

theonesean commented:
After finally configuring autosave on Atom (and regretting not using emacs for this project thus far), I've implemented list-mode and cleaned up my code. D is really great.
